tests/subsys/debug/coredump/src/main.c

```diff
@@ -25,12 +25,18 @@ void k_sys_fatal_error_handler(unsigned int reason, const z_arch_esf_t *pEsf)
 	k_fatal_halt(reason);
 }
 
-void func_3(uint32_t *addr)
+/* Turn off optimizations to prevent the compiler from optimizing this away
+ * due to the null pointer dereference.
+ */
+__no_optimization void func_3(uint32_t *addr)
```

Comment:
I don't think you need `__no_optimization` if it is going to trigger `k_panic()`.

Reply:
`__no_optimization` is still needed for boards that won't go through `k_panic()` when building with clang (`qemu_x86` and `qemu_riscv32/64`, for example--these are where I first bumped into this). The boards I added to the `#if` are just those that were negatively impacted by `__no_optimization` being added (I added them to fix the CI check for this PR).

As an alternative, I think I could drop the `__no_optimization` and add a change to always take the `k_panic()` route when using clang, but I'm not sure if that is better.

Reply:
In that case, please split this into two commits. One is to add the #ifdef for RISCV board, and the other one adding `__no_optimization`.

Reply:
I have now split this into two separate commits!